docs/topic-guides/workflows/workflow-file.rst

```diff
@@ -0,0 +1,362 @@
+``$outputs``, and ``$parameters`` templates to tell Renku "put all the inputs
+here".
+
+.. code-block:: yaml
+
+    name: flights-processing-pipeline
+    steps:
+      filter:
+        command: python $parameters $inputs $outputs
+        inputs:
+          - src/filter_flights.py
+          - data/flight-data/2019-01-flights.csv.zip
+        outputs:
+          - data/output/flights-filtered.csv
+        parameters:
+          - n
+          - 10
+
+
+The inputs are filled in to the command at the ``$inputs`` template in the order
+in which they are specified in the ``inputs`` section.
+
+
+Templating by Argument Name
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the ordering of arguments in your command is more complex, you can reference
+each argument individually by name. To do so, assign each input and output a
+name (such as ``raw-flights``) and a ``path``. Then, we reference those names in
+the ``command`` using ``$``.
+
+.. code-block:: yaml
+
+    name: flights-processing-pipeline
+    steps:
+      filter:
+        command: python $n $filter-py $raw-flights $filtered-flights
+        inputs:
+          - filter-py:
+              path: src/filter_flights.py
+          - raw-flights:
+              path: data/flight-data/2019-01-flights.csv.zip
+        outputs:
+          - filtered-flights:
+              path: data/output/flights-filtered.csv
+        parameters:
+          - n:
+            prefix: -n
+            value: 10
+
+.. note:: If your command uses the ``$`` character, you can escape it by doing ``$$``.
+
+
+A Multi-Step Workflow File
+--------------------------
+
+Below, you can see what the a workflow file looks like for a two-step
+workflow.
+
+.. code-block:: yaml
+
+    name: flights-processing-pipeline
+    steps:
+      filter:
+        command: python $filter-py $raw-flights $filtered-flights
+        inputs:
+          - filter-py:
+              path: src/filter_flights.py
+          - raw-flights:
+              path: data/flight-data/2019-01-flights.csv.zip
+        outputs:
+          - filtered-flights:
+              path: data/output/flights-filtered.csv
+
+      count:
+        command: python $count-py $filtered-flights $flight_count
+        inputs:
+          - count-py:
+              path: src/count_flights.py
+          - filtered-flights:
+              path: data/output/flights-filtered.csv
+        outputs:
+          - flight_count:
+              path: data/output/flights-count.csv
+
+
+Executing a Workflow File
+-------------------------
+
+Running :meth:`renku run workflow.yml <renku.ui.cli.run>` will execute all steps
+in the workflow file. Executing the workflow will commit all workflow inputs and
+outputs, too, including the workflow file itself.
+
+.. code-block:: console
+
+    $ renku run workflow.yml
+    Executing step 'workflows/workflow-topic-guide-last.yml::filter': 'python src/filter_flights.py data/flight-data/2019-01-flights.csv.zip data/output/flights-filtered.csv' ...
```

Comment:
As we discussed today, the name will change to `flights-processing-pipeline::filter`. This is not final yet since I need to decide about the separator to be `::` or `.`.

Reply:
Updated to `.`